src/panels/config/integrations/integration-panels/thread/thread-config-panel.ts

```diff
@@ -467,10 +487,7 @@ export class ThreadConfigPanel extends SubscribeMixin(LitElement) {
     const network = (ev.currentTarget as any).network as ThreadNetwork;
     const router = (ev.currentTarget as any).router as ThreadRouter;
     const otbr = (ev.currentTarget as any).otbr as OTBRInfo;
-    const index =
-      network.dataset && router.border_agent_id
-        ? Number(ev.detail.index)
-        : Number(ev.detail.index) + 1;
+    const index = Number(ev.detail.index);
```

Comment:
So we are sure we always have a dataset for OTBR networks right?

Reply:
Pretty much, yes: The config flow of the OTBR integration makes sure that the OTBR has an active dataset: If there is a preferred dataset with a TLV in our store, it joins that network, otherwise it creates a new one. The active dataset is then added to the Thread store on OTBR config entry setup.